crates/cxx-qt-build/src/lib.rs

```diff
@@ -251,12 +239,86 @@ fn generate_cxxqt_cpp_files(
 }
 
 fn panic_duplicate_file_and_qml_module(
-    path: &Path,
+    path: impl AsRef<Path>,
     uri: &str,
     version_major: usize,
     version_minor: usize,
 ) {
-    panic!("CXX-Qt bridge Rust file {} specified in QML module {uri} (version {version_major}.{version_minor}), but also specified via CxxQtBuilder::file. Bridge files must be specified via CxxQtBuilder::file or CxxQtBuilder::qml_module, but not both.", path.display());
+    panic!("CXX-Qt bridge Rust file {} specified in QML module {uri} (version {version_major}.{version_minor}), but also specified via CxxQtBuilder::file. Bridge files must be specified via CxxQtBuilder::file or CxxQtBuilder::qml_module, but not both.", path.as_ref().display());
+}
+
+/// Metadata for registering a QML module with [CxxQtBuilder::qml_module]
+pub struct QmlModule<'a, A, B>
```

Comment:
very minor comment, these structs could be put in their own file.

Elsewhere where we have been doing one struct / topic per Rust file, it has helped reduce having one massive file :-) And makes them easier to find etc

Reply:
done